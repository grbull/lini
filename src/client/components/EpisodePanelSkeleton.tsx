import cn from 'classnames';
import React, { ReactElement } from 'react';

import { Skeleton } from './Skeleton';

export function EpisodePanelSkeleton(): ReactElement {
  return (
    <div
      className={cn('w-full', 'overflow-x-auto', 'whitespace-nowrap', 'px-1')}
    >
      {[1, 2, 3, 4].map((i) => (
        <div className={cn('inline-block', 'w-1/3', 'p-1')} key={i}>
          <Skeleton
            className={cn(
              'w-full',
              'h-0',
              'relative',
              'bg-gray-600',
              'rounded-sm'
            )}
            style={{ paddingTop: `${(295 / 210) * 100}%` }}
          />
          <Skeleton className={cn('h-18', 'bg-gray-800')} />
        </div>
      ))}
    </div>
  );
}
