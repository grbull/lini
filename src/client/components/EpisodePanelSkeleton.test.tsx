/**
 * @jest-environment jsdom
 */

import '@testing-library/jest-dom/extend-expect';

import { render } from '@testing-library/react';
import React from 'react';

import { EpisodePanelSkeleton } from './EpisodePanelSkeleton';

describe('EpisodePanelSkeleton Component', () => {
  it('matches the snapshot', () => {
    const { asFragment } = render(<EpisodePanelSkeleton />);
    expect(asFragment()).toMatchSnapshot();
  });
});
