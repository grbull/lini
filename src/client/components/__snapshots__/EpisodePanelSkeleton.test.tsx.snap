// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpisodePanelSkeleton Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="w-full overflow-x-auto whitespace-nowrap px-1"
  >
    <div
      class="inline-block w-1/3 p-1"
    >
      <div
        class="animate-pulse bg-gray-700 w-full h-0 relative bg-gray-600 rounded-sm"
        style="padding-top: 140.47619047619045%;"
      />
      <div
        class="animate-pulse bg-gray-700 h-18 bg-gray-800"
      />
    </div>
    <div
      class="inline-block w-1/3 p-1"
    >
      <div
        class="animate-pulse bg-gray-700 w-full h-0 relative bg-gray-600 rounded-sm"
        style="padding-top: 140.47619047619045%;"
      />
      <div
        class="animate-pulse bg-gray-700 h-18 bg-gray-800"
      />
    </div>
    <div
      class="inline-block w-1/3 p-1"
    >
      <div
        class="animate-pulse bg-gray-700 w-full h-0 relative bg-gray-600 rounded-sm"
        style="padding-top: 140.47619047619045%;"
      />
      <div
        class="animate-pulse bg-gray-700 h-18 bg-gray-800"
      />
    </div>
    <div
      class="inline-block w-1/3 p-1"
    >
      <div
        class="animate-pulse bg-gray-700 w-full h-0 relative bg-gray-600 rounded-sm"
        style="padding-top: 140.47619047619045%;"
      />
      <div
        class="animate-pulse bg-gray-700 h-18 bg-gray-800"
      />
    </div>
  </div>
</DocumentFragment>
`;
